pkg/kubectl/cmd/annotate.go

```diff
@@ -128,10 +128,11 @@ func NewCmdAnnotate(f cmdutil.Factory, out io.Writer) *cobra.Command {
 		ArgAliases: argAliases,
 	}
 	cmdutil.AddPrinterFlags(cmd)
+	cmdutil.AddIncludeUninitializedFlag(cmd)
 	cmd.Flags().Bool("overwrite", false, "If true, allow annotations to be overwritten, otherwise reject annotation updates that overwrite existing annotations.")
 	cmd.Flags().Bool("local", false, "If true, annotation will NOT contact api-server but run locally.")
-	cmd.Flags().StringP("selector", "l", "", "Selector (label query) to filter on, supports '=', '==', and '!='.(e.g. -l key1=value1,key2=value2)")
-	cmd.Flags().Bool("all", false, "Select all resources in the namespace of the specified resource types")
+	cmd.Flags().StringP("selector", "l", "", "Selector (label query) to filter on, supports '=', '==', and '!='.(e.g. -l key1=value1,key2=value2). Does not include the uninitialized objects by default, unless user explicitly set --include-uninitialized=true.")
+	cmd.Flags().Bool("all", false, "Select all resources (including the uninitialized objects by default, unless user explicitly set --include-uninitialized=false. And this will be override when --selector is specified) in the namespace of the specified resource types.")
```

Comment:
A discussion about uninitialized objects (the details of how this works) belongs in the description section of the command, not in the flags.  Flag descriptions need to be short sentence.  I would add a sentence or two to the description of annotate and other commands rather than adding this very long description in the flag.

Reply:
Current the descriptions of some subcommands are quite long enough, like `kubectl delete`. I'm afraid new added items will not get noticed.

Both will be okay. @caesarxuchao any comments?

Reply:
I think adding to the command description is better. After all, `--all` include unintialized objects by default is intuitive.